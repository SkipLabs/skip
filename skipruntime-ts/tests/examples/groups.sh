#!/bin/bash
set -e
cd ../../examples

if [ "$#" -ne 2 ]; then
    echo "Usage: $0 <stdout target> <stderr target>"
fi

node dist/groups.js >/dev/null &
node dist/groups-server.js >/dev/null &

sleep 1 # give a moment for service to spin up
node dist/groups-client.js >"$1" 2>"$2"

jobs -p | xargs kill
