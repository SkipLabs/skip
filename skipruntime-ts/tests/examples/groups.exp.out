Listening for Bob's active friends in each group...
Initial data:  [ [ 1001, [ 1 ] ] ]
Setting Carol to active...
Updated data:  [ [ 1001, [ 1, 2 ] ], [ 1002, [ 2 ] ] ]
Setting Alice to inactive...
Updated data:  [ [ 1001, [ 2 ] ] ]
Setting Eve as Bob's friend...
Updated data:  [ [ 1001, [ 2, 3 ] ], [ 1002, [ 2 ] ] ]
Removing Carol and adding Eve to group 2...
Updated data:  [ [ 1002, [ 3 ] ] ]
Closing listener event stream...
